tests/server/test_resource_details.py

```diff
@@ -0,0 +1,327 @@
+    assert result.code == 200
+    assert result.result["data"]["first_generated_version"] == 2
+    generated_time = datetime.datetime.strptime(result.result["data"]["first_generated_time"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert generated_time == cm_times[1].astimezone(datetime.timezone.utc)
+    deploy_time = datetime.datetime.strptime(result.result["data"]["last_deploy_finished"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert deploy_time == resources[env.id][multiple_requires][3].last_deploy.astimezone(datetime.timezone.utc)
+    assert result.result["data"]["attributes"] == resources[env.id][multiple_requires][3].attributes
+    assert result.result["data"]["requires_status"] == {
+        "std::Directory[internal,path=/tmp/dir1],v=4": "deployed",
+        "std::File[internal,path=/tmp/dir1/file2],v=4": "deploying",
+    }
+    assert result.result["data"]["status"] == "deployed"
+
+    no_requires = ids["no_requires"]
+    result = await client.resource_details(env.id, no_requires)
+    assert result.code == 200
+    assert result.result["data"]["first_generated_version"] == 2
+    generated_time = datetime.datetime.strptime(result.result["data"]["first_generated_time"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert generated_time == cm_times[1].astimezone(datetime.timezone.utc)
+    deploy_time = datetime.datetime.strptime(result.result["data"]["last_deploy_finished"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert deploy_time == resources[env.id][no_requires][3].last_deploy.astimezone(datetime.timezone.utc)
+    assert result.result["data"]["attributes"] == resources[env.id][no_requires][3].attributes
+    assert result.result["data"]["requires_status"] == {}
+    assert result.result["data"]["status"] == "deployed"
+
+    single_requires = ids["single_requires"]
+    result = await client.resource_details(env.id, single_requires)
+    assert result.code == 200
+    assert result.result["data"]["first_generated_version"] == 4
+    generated_time = datetime.datetime.strptime(result.result["data"]["first_generated_time"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert generated_time == cm_times[3].astimezone(datetime.timezone.utc)
+    deploy_time = datetime.datetime.strptime(result.result["data"]["last_deploy_finished"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+        tzinfo=datetime.timezone.utc
+    )
+    assert deploy_time == resources[env.id][single_requires][3].last_deploy.astimezone(datetime.timezone.utc)
+    assert result.result["data"]["attributes"] == resources[env.id][single_requires][3].attributes
+    assert result.result["data"]["requires_status"] == {"std::Directory[internal,path=/tmp/dir1],v=4": "deployed"}
+    assert result.result["data"]["status"] == "deploying"
```

Comment:
I would expect the state to be orphaned here?

Reply:
The latest released version is 4, so it's not orphaned, but the orphaned case was missing, added it